core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -77,6 +80,10 @@ public boolean isForceExecution() {
         return forceExecution;
     }
 
+    public boolean isCheckSizeLimit() {
```

Comment:
I agree that the old name is not ideal but I think `checkInflightRequestLimit` is suitable here. I still think this name should not need to change in case we'd implement the check differently (not that we want to do this, it's just a thought experiment to make clear that `canTripCircuitBreaker` is implementation specific. Even in that case I'd go so far as to call it `canTripInFlightRequestCircuitBreaker`).